traces/stf_trace_gen/dromajo_stf_lib.patch

```diff
@@ -81,212 +213,107 @@ index 2f71c99..2de97a3 100644
-+                    }
-+
-+                    if(inst_width == 4) {
-+                        stf_writer << stf::InstOpcode32Record(insn_raw);
-+                    }
-+                    else {
-+                        stf_writer << stf::InstOpcode16Record(insn_raw & 0xFFFF);
-+                    }
-+                }
-+            }
-+        }
-+        return keep_going;
-+    }
-+
-     if (!do_trace) {
-         return keep_going;
-     }
-@@ -223,7 +322,7 @@ int main(int argc, char **argv) {
-     if (!m)
-         return 1;
- 
--    int n_cycles = 10000;
-+    int n_cycles = 1;
-     execution_start_ts = get_current_time_in_seconds();
-     execution_progress_meassure = &m->cpu_state[0]->minstret;
-     signal(SIGINT, sigintr_handler);
+ #ifdef GOLDMEM_INORDER
+ void check_inorder_load(int cid, uint64_t addr, uint8_t sz, uint64_t ld_data, bool io_map);
+ void check_inorder_store(int cid, uint64_t addr, uint8_t sz, uint64_t st_data, bool io_map);
 diff --git a/src/dromajo_main.cpp b/src/dromajo_main.cpp
-index aeb1912..2fdaef2 100644
+index aeb1912..3369f79 100644
 --- a/src/dromajo_main.cpp
 +++ b/src/dromajo_main.cpp
 @@ -560,6 +560,7 @@ static void usage(const char *prog, const char *msg) {
              "       --maxinsns terminates execution after a number of instructions\n"
              "       --terminate-event name of the validate event to terminate execution\n"
              "       --trace start trace dump after a number of instructions. Trace disabled by default\n"
-+	     "       --stf_trace <filename>  Dump an STF trace to the given file\n"
++            "       --stf_trace <filename>  Dump an STF trace to the given file\n"
              "       --ignore_sbi_shutdown continue simulation even upon seeing the SBI_SHUTDOWN call\n"
              "       --dump_memories dump memories that could be used to load a cosimulation\n"
              "       --memory_size sets the memory size in MiB (default 256 MiB)\n"
-@@ -618,6 +619,7 @@ RISCVMachine *virt_machine_main(int argc, char **argv) {
+@@ -571,6 +572,7 @@ static void usage(const char *prog, const char *msg) {
+             "       --plic START:SIZE set PLIC start address and size in B (defaults to 0x%lx:0x%lx)\n"
+             "       --clint START:SIZE set CLINT start address and size in B (defaults to 0x%lx:0x%lx)\n"
+             "       --custom_extension add X extension to misa for all cores\n"
++            "       --no_stf_priv_check Turn off the privledge check in STF generation\n"
+ #ifdef LIVECACHE
+             "       --live_cache_size live cache warmup for checkpoint (default 8M)\n"
+ #endif
+@@ -618,10 +620,12 @@ RISCVMachine *virt_machine_main(int argc, char **argv) {
      long        ncpus                    = 0;
      uint64_t    maxinsns                 = 0;
      uint64_t    trace                    = UINT64_MAX;
-+    const char *stf_trace                = nullptr;
++    const char *stf_trace                = NULL;
```

Comment:
Why change this to `NULL`?  `nullptr` is preferred simply for the fact it's a type.  `NULL` is just a `#define` which can be set to anything...

Reply:
i copied the style found in the original source.

e.g. const char *cmdline  = NULL; //as in dromajo_main.cpp

I will change it to nullptr